### PR TITLE
Move Api Actions to separate package and fix type issues

### PR DESCRIPTION
**Pull request by Oksamies**

### CODE REVIEW

**Comment by Oksamies** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1033?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `master`
  * **PR #1023** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1023?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #1024** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1024?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #1029** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1029?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #1030** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1030?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #1031** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1031?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #1032** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1032?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #1033** <a href="https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1033?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/thunderstore-io/thunderstore-ui/1033?utm_source=stack-comment).

File: packages/ts-api-react-actions/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# ts-api-react-actions
+
+React actions logic for the Thunderstore API
```
Comment (anttimaki) [inline packages/ts-api-react-actions/README.md:3]: This tells me nothing TBH

**Review by anttimaki — COMMENTED** [review]
LGTM but this was a cursory review for the changes as they are, without giving any thought on whether they make sense in the big picture.

File: packages/ts-api-react-actions/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  "dependencies": {
+    "@thunderstore/thunderstore-api": "*",
+    "@types/react": "^18.0.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "zod": "^3.22.2"
+  },
+  "devDependencies": {
+    "typescript-plugin-css-modules": "^3.4.0"
```
[the remark this answers is not in the document]
Comment (Oksamies) [reply packages/ts-api-react-actions/package.json:24]: Probably not, did a quick copypaste of the forms package. I'll clean this up.

**Pull request merged**